src/pip/_internal/req/req_install.py

```diff
@@ -595,9 +595,10 @@ def run_egg_info(self):
                 'Running setup.py (path:%s) egg_info for package from %s',
                 self.setup_py_path, self.link,
             )
-        base_cmd = make_setuptools_shim_args(self.setup_py_path)
-        if self.isolated:
-            base_cmd += ["--no-user-cfg"]
+        base_cmd = make_setuptools_shim_args(
+            self.setup_py_path,
+            no_user_cfg=self.isolated
```

Comment:
Since we're not limited by backwards compatibility, we might as well write this out fully to have a nicer name: `no_user_config`.